cmd/logger.go

```diff
@@ -137,7 +92,78 @@ func getSource() string {
 	return fmt.Sprintf("[%s:%d:%s()]", filename, lineNum, funcName)
 }
 
-func logIf(level logrus.Level, source string, err error, msg string, data ...interface{}) {
+func printfLevel(level, msg, time, cause,
+	source string, stack []string, data ...interface{}) {
+
+	stackTab := "       "
+	levelCaps := strings.ToUpper(level)
```

Comment:
Why aren't we using golang `log` package?

Reply:
Could you please elaborate what advantages we'll gain by switching to `log` library package?
As far as I can tell and understand, our need is very basic and fundamental, hence we don't need all the bells and whistles `log` library package brings in, like Panicln, Panicf, Fatalf, Flags, Output etc.
If you'd like, we can discuss it further with AB and the team about what advantage(s) switching to the `log` library will bring in and then we can go with it if we decide to.

Reply:
@abperiasamy also reviewed the log library package and  says that log library does not offer what we need and it doesn't even make sense to extend on top of it.

Reply:
:+1:

Reply:
In such a scenario it is worthwhile to consider making this part of the code a log package.

Reply:
@harshavardhana we do plan to make this a log package. We are trying to make incremental changes. Once this PR is merged, we plan to have three more PRs one of which will be exclusively about making logging a separate package.

Reply:
More review comments from @abperiasamy :
* Introduce a new `Init` function for `TracePoint` and set `rootPath` in there only once.
* Instead of mixing built-in stack trace and our stack trace (when built-in is not available), we'll start using ours all the time, for there is no benefit of mixing them and to make stack trace log information consistent all around.
* Rename `Stack: ` as `Trace: ` and display the stack information in reverse order
* Reorder the fields in the log output as:
    * Trace
    * Source
    * Time
    * Cause
    * Error/Fatal message